src/glang/codegen/translation_unit.py

```diff
@@ -68,7 +68,13 @@ def _add_parameter(self, param_name: str, param_type: Type):
         self._parameters.append(fn_param)
         self.top_level_scope.add_generatable(fn_param)
 
-        fn_param_var = StackVariable(param_name, param_type, False, True)
+        # Allow assignments to parameters passed by value (but our references
+        # still need to be immutable).
+        is_value_type = param_type.storage_kind == Type.Kind.VALUE
+
+        fn_param_var = StackVariable(
+            param_name, param_type, is_mutable=is_value_type, initialized=True
+        )
```

Comment:
Fair. This breaks the const-by-default assumption but it also avoids redundant copies when you actually need to mutate.

Reply:
I do like the const-by-default, but we don't really have syntax for mutable arguments. Maybe we should